adbc.h

```diff
@@ -152,15 +152,15 @@ struct ArrowArrayStream {
 // Storage class macros for Windows
 // Allow overriding/aliasing with application-defined macros
 #if !defined(ADBC_EXPORT)
-#if defined(_WIN32)
+#if defined(_MSC_VER)
```

Comment:
We need to use `_WIN32` not `_MSC_VER` here because MinGW doesn't define `_MSC_VER` but it requires `__delcspec(...)`. `_WIN32` is defined on both of MinGW and MSVC.